src/idl_gen_cpp.cpp

```diff
@@ -1506,10 +1502,8 @@ class CppGenerator : public BaseGenerator {
         if (ev.IsZero()) { continue; }
 
         code_.SetValue("LABEL", GetEnumValUse(enum_def, ev));
-        code_.SetValue("TYPE",
-                       GetUnionElement(ev, true, true, true, true, opts_));
-        code_.SetValue("NAME",
-                       GetUnionElement(ev, false, true, false, false, opts_));
+        code_.SetValue("TYPE", GetUnionElement(ev, true, true, opts_));
+        code_.SetValue("NAME", GetUnionElement(ev, false, false, opts_));
```

Comment:
This line is equal to (tested with code before this PR):
```cpp
(ev.union_type.base_type == BASE_TYPE_STRUCT) ? ev.union_type.struct_def->name : "flatbuffers::String";
```
But the resulted "Name=flatbuffers::String" is never used (L1516-L1517).
I think we can simply write
```
 code_.SetValue("NAME",
     (ev.union_type.base_type == BASE_TYPE_STRUCT) ? ev.union_type.struct_def->name : "");
```
and remove the second argument from `GetUnionElement()`.

Reply:
Even simpler: move `SetValue("NAME",...)` into the scope where it's actually needed.